app/src/main/scala/org/alephium/explorer/service/TransactionService.scala

```diff
@@ -244,39 +235,34 @@ object TransactionService extends TransactionService {
                                 batchSize: Int,
                                 paralellism: Int)(
       implicit ec: ExecutionContext,
-      dc: DatabaseConfig[PostgresProfile]): Source[ArraySeq[Transaction], NotUsed] = {
-    Source
+      dc: DatabaseConfig[PostgresProfile]): Flowable[ArraySeq[Transaction]] = {
+    Flowable
       .fromPublisher(stream(streamTxByAddressQR(address, from, to)))
-      .grouped(batchSize)
-      .mapAsync(paralellism) { hashes =>
-        run(getTransactionsNoJoin(ArraySeq.from(hashes)))
-      }
+      .buffer(batchSize)
+      .concatMapEager(({ hashes =>
+        Flowable.fromCompletionStage(
+          run(getTransactionsNoJoin(ArraySeq.from(hashes.asScala))).asJava
+        )
+      }), paralellism, 1)
```

Comment:
So that the `Stream` handles back-pressure more effectively, we could consider replacing the mid-stream query execution using `run(getTransactionsNoJoin ...`) with a `Publisher` that is returned without executing a query? For example:

```scala
Flowable
  .fromPublisher(stream(streamTxByAddressQR(address, from, to)))
  .buffer(batchSize)
  .flatMap(hashes => stream(getTransactionsNoJoin_Streamed(ArraySeq.from(hashes.asScala))))
```

`getTransactionsNoJoin` would need to be stream-fied for this.

Reply:
Totally agree, I'll check if it's possible to convert the query into a stream....Slick can be a pain sometimes.....

Reply:
I had a quick look and it will be more complex than expected, because [here](https://github.com/alephium/explorer-backend/blob/master/app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala#L335-L341) we need to get all inputs/outputs and it's a non-streaming action. Then Slick doesn't love it and it seems we can't go back to a streaming.  But I'll open an issue for that, because it would be definitively cool to have a full reactive pipe from the http request up to the response.